### PR TITLE
Fixed OCR2DR Oracle events to match o11y needs

### PR DESCRIPTION
**Pull request by pinebit**
For o11y to work properly, requestID needs to be indexed. Also adding subscriptionId parameter to make it a foreign key to match future billing o11y records.

### CODE REVIEW

File: contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol [hunk 1]
```diff
@@ -9,10 +9,10 @@ import "../ocr2/OCR2Base.sol";
  * @dev THIS CONTRACT HAS NOT GONE THROUGH ANY SECURITY REVIEW. DO NOT USE IN PROD.
  */
 contract OCR2DROracle is OCR2DROracleInterface, OCR2Base {
-  event OracleRequest(bytes32 requestId, bytes data);
-  event OracleResponse(bytes32 requestId);
-  event UserCallbackError(bytes32 requestId, string reason);
-  event UserCallbackRawError(bytes32 requestId, bytes lowLevelData);
+  event OracleRequest(bytes32 indexed requestId, uint64 subscriptionId, bytes data);
```
Comment (bolekk) [inline contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol:12]: Do you have some vision for other data that we might need to add here in the future?
https://app.shortcut.com/chainlinklabs/story/58066/metadata-structure
https://smartcontract-it.atlassian.net/browse/DRO-4

I'm OK with having a top-level subscriptionID because it's a fundamental attribute and will be immediately useful for the new function management logic in the EA. But let's come up with a plan before adding more and more stuff (e.g. do we want to have a sub-structure? how does it effect the topic hash of the event (we'd prefer that it doesn't change once we start collecting real data). What goes inside CBOR-encoded "data" vs directly here?).

**Review by bolekk — APPROVED** [review]

File: contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol (hunk 1 above)
Comment (pinebit) [reply contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol:12]: There is the current vision for PoC/MVP that was presented here some time ago: https://www.notion.so/chainlink/OCR2-Direct-Request-Observability-0aa7187203b8486abd22905148cc7e45
I added subscriptionId for two reasons:

- to easily match the future events of the billing registry,
- to be able to pass that property all the way down to EA.

I would not add any other data to the events until it is absolutely required. In that case we will either introduce an additional event (to not break backward compatibility), or we will change the topic in o11y path.
To sum up, at this point I do not see any other properties that we should add here, to this specific event of this specific contract.

Comment (bolekk) [reply contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol:12]: There are 3 properties mentioned in the stories I linked above:
1. Client's commitment object
2. Aggregation method flag
3. Secret's signer

Comment (pinebit) [reply contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol:12]: > What goes inside CBOR-encoded "data" vs directly here?
`data` would require decoding and this is difficult to be done as a part of SQL queries in Atlas/Grafana. I am inclining to keep data as CBOR blob which we can decode for a given request if needed during a debugging session.

Comment (bolekk) [reply contracts/src/v0.8/dev/ocr2dr/OCR2DROracle.sol:12]: Sure. Another key differentiator for what goes inside vs outside "data" is whether it comes from the client contract vs is added by the oracle contract.
